components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -101,6 +102,7 @@ public class IdentityProviderManager implements IdpManager {
     private static final String OPENID_IDP_ENTITY_ID = "IdPEntityId";
     private static CacheBackedIdPMgtDAO dao = new CacheBackedIdPMgtDAO(new IdPManagementDAO());
     private static volatile IdentityProviderManager instance = new IdentityProviderManager();
+    private static Map<String, List<String>> metaFedAuthConfigMap = new HashMap<>();
```

Comment:
This should not be a static map.
This is not thread safe. If two tenants access the IDPs there is a change this might be modified unintentionally.

Reply:
Also this is a singleton class. So we having a static map is not safe as same instance will be used for all tenants